### ISSUE TITLE
v27.0 Testing

### ISSUE THREAD

**Issue by marcofleon** [bitcoin/bitcoin#29697]
Umbrella issue for 27.0 testing. Please help testing on a wide variety of supported platforms, as well as interaction with different software.

Let us know which version you tested on which operating system.

If you find an issue, please search Github for known issues first and then open a new Github issue.

_This meta issue should not be used to report bugs, as a single thread makes it impossible to track more than one topic._

See [27.0 Release Notes Draft](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Notes-Draft) for a list of changes.

See here for the testing guide: https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide

**Comment by jonasnick** [comment]
Tested 27.0rc1 on NixOS without encountering issues using a variant of this script: https://gist.github.com/erikarvstedt/62c38855e0092cd4adffa83cf751023b.

The script runs the [nix-bitcoin test framework](https://github.com/fort-nix/nix-bitcoin/blob/master/test/tests.py) which starts bitcoind and a bunch of dependent services (CLN, elementsd, LND, joinmarket, mempool, fulcrum, btcpayserver, electrs) and checks whether they started up correctly.

**Comment by cbergqvist** [comment]
@jonasnick 

What's the status of the disabling of JoinMarket?
https://gist.github.com/erikarvstedt/62c38855e0092cd4adffa83cf751023b#file-test-bitcoind-26-sh-L26-L27

**Comment by cbergqvist** [comment]
While going through the Testing Guide's [v3 transaction section](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide#v3-transaction-policy) I noticed that if one attempts to do a `testmempoolaccept` just before the last `submitpackage`, the former would reject the input, but the latter would still allow it.
```console
$ bcli testmempoolaccept "[\"$v3_parent_signed_tx\",\"$high_fee_child_signed_tx\"]"
[
  {
    "txid": "e40a9567347a3eace38f6e8c05d437c1c56460ad34ddeed724e595f776e16bff",
    "wtxid": "06551611e1c23087abd4f72d928c55ec69cf30cb19f5ffa7af00a87b8d77ad40",
    "allowed": false,
    "reject-reason": "txn-already-in-mempool"
  },
  {
    "txid": "66c8a8dfa2ba6995caae3dd1f109ea37408dd65f0a94cc2cd0872a4a5604a2a9",
    "wtxid": "0a67d798ad3f8be6875dfaa562cc1f2dd15b70e368c915cdca0e66e2a686c952"
  }
]
```

**Comment by glozow** [comment]
> While going through the Testing Guide's [v3 transaction section](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/27.0-Release-Candidate-Testing-Guide#v3-transaction-policy) I noticed that if one attempts to do a `testmempoolaccept` just before the last `submitpackage`, the former would reject the input, but the latter would still allow it.

Yes, they have slightly different interfaces, for various reasons. For example
- Your example; `testmempoolaccept` errors when the tx is already there. I didn't change this when adding multi-tx functionality because it would be a change to an interface that had existed for a while. For `submitpackage`, I felt it would be much more useful to be tolerant of already-in-mempool transactions.
- We aren't really able to simulate package CPFPs in test-accepts at the moment (or at least it's a bit more difficult) so `testmempoolaccept` will fail on those

This isn't a bug, though I can see why it's confusing for people. Perhaps we can add more documentation about this.

**Comment by 0xB10C** [comment]
fwiw: https://github.com/bitcoin/bitcoin/issues/29785

**Comment by laanwj** [comment]
> What's the status of the disabling of JoinMarket?

joinmarket (master) works with v27.0rc1, both sendpayment and the yield generator
